tlscanary/template/js/index_page.js

```diff
@@ -0,0 +1,346 @@
+      type: null,
+      width: "20%"
+    }
+  );
+  // Add an invisible column which allows one field to become the identifier
+  columns.unshift(
+    {
+      name: "id",
+      default: false,
+      type: "str",
+      width: "0%"
+    }
+  );
+
+  var html = "<table id='grid' class='table table-condensed table-hover table-striped'><thead><tr>";
+  for (var i = 0;i < columns.length;i++) {
+    html += "<th data-column-id='" + columns[i].name + "' ";
+    if (columns[i].name === "id")
+    {
+      html += "data-visible-in-selection='false'  data-searchable='false' ";
+      html += "data-identifier='true' ";
+    } 
+    if (columns[i].name === "Actions")
+    {
+      html += "data-visible-in-selection='false' data-formatter='commands' data-searchable='false' ";
+    }
+    if (columns[i].name === "run")
+    {
+      html += "data-order='desc' data-formatter='link' "; 
+    }
+    if (!columns[i].default)
+    {
+      html += "data-visible='false' ";
+    }
+    if (columns[i].type === "int")
+    {
+      html += "data-type='numeric' ";
+    }
+    if (typeof(columns[i].width) !== "undefined")
+    {
+      html += "data-width='" + columns[i].width + "' ";
+    } else {
+      html += "data-width='20%' ";
+    }
+    html += ">" + columns[i].name + "</th>"
+  }
+  html += "</tr></thead><tbody>";
+  for (var i = 0;i < hosts.length;i++) {
+    html += "<tr id=\'" + hosts[i]["run"] + "\'>";
+    html += "<td>" + hosts[i]["run"] + "</td>"
+    for (var j = 1;j < columns.length;j++)
+    {
+      html += "<td>" + hosts[i][columns[j].name] + "</td>"
+    }
+    html += "</tr>";
+  }
+  html += "</tbody></table>";
+  const contentDiv = document.getElementById("results");
+  contentDiv.style.visibility = "hidden";
+  contentDiv.innerHTML = html;
+}
+
+function applyBootgrid() {
+  const grid = $("#grid").bootgrid(
+  {
+    css:
+    {
+      paginationButton: "labels_small"
+    },
+    rowCount: [10,5,-1],
+    selection: false,
+    multiSelect: true,
+    rowSelect: true,
+    keepSelection: true,
+    formatters: 
+    {
+      "commands": function(column, row) {
+        var html = "";                  
+        html += "<button type='button' title='Link' class='btn btn-xs btn-default command-link' data-row-id='runs/"
+             + row.id + "/index.htm'><span>&#128279; </span></button> " +
+              "<button type='button' title='Remove from list' class='btn btn-xs btn-default command-delete' data-row-id='"
+             + row.id + "'><span class='fa fa-trash-o'> &times; </span></button>";
+        return html;
+      },
+      "date": function(column, row) {
+        var temp = String(row[column.id]).substr(0,13);
+        var d = new Date(Number(temp));
+        return d.toString();
+      },
+      "link": function(column, row) {
+        var html = "<a href='runs/" + row.id + "/index.htm' target='_blank'>" + row.id + "</a>";
+        return html;
+      }
+    }
+  }).on("loaded.rs.jquery.bootgrid", function ()
+  {
```

Comment:
Curly without linebreak? (as above)

Reply:
There are other places all over the code. I won't denote them all. Feel free to address them as you see fit.